### PR TITLE
83 create a correction overview view

### PR DESCRIPTION
**Pull request by JanKneiphof**
This adds an overview for corrections to the app.
You can toggle between the corrected answers of your user account and the corrections you gave to the answers of other users.
The paging/tabs are solved with get Parameters + Bootstrap, so no additional javascript is used.
Screenshot from 2023-08-11 12-14-25

### CODE REVIEW

File: src/clj/db.clj [hunk 1]
```diff
@@ -494,7 +502,47 @@
...
+             [?correction :correction/answer ?answer]
+             [?correction :correction/feedback ?feedback ?tx]
+             [?answer :answer/user ?user]
+             [?answer :answer/points ?points-reached]
+             [?answer :answer/answer ?answers]
+             [?answer :answer/question ?question]
+             [?question :question/question-statement ?question-statement]
+             [?question :question/points ?reachable-points]
+             [?tx :db/txInstant ?timestamp]
+             [?user :user/id ?user-id]]
```
Comment (hehin101) [inline src/clj/db.clj:524]: I am not sure if this is the best order.

File: src/clj/db.clj [hunk 2]
```diff
@@ -494,7 +502,47 @@
...
+             [?correction :correction/answer ?answer]
+             [?correction :correction/feedback ?feedback ?tx]
+             [?correction :correction/corrector ?user]
+             [?answer :answer/points ?points-reached]
+             [?answer :answer/answer ?answers]
+             [?answer :answer/question ?question]
+             [?question :question/question-statement ?question-statement]
+             [?question :question/points ?reachable-points]
+             [?tx :db/txInstant ?timestamp]
+             [?user :user/id ?user-id]]
```
Comment (hehin101) [inline src/clj/db.clj:544]: Same here. Not sure if there is a better order for this.

File: src/clj/views/correction/correction_overview_view.clj
Comment (hehin101) [inline src/clj/views/correction/correction_overview_view.clj:1]: Missing specs and tests for `correction-overview`.

**Review by hehin101 — CHANGES_REQUESTED** [review]
Very good.

A little bit confusing:
I can still start a "next page" even if there are no more corrections left.
That allows me to cycle through empty pages.

Requested changes:
The missing specs and tests have to be added.

File: src/clj/views/correction/correction_overview_view.clj
Comment (JanKneiphof) [reply src/clj/views/correction/correction_overview_view.clj:1]: I added them now for `correction-overview`

File: src/clj/db.clj (hunk 2 above)
Comment (JanKneiphof) [reply src/clj/db.clj]: I completely forgot to think about the order on these queries, thank you for the hint :D
I updated them, what do you think about them now?

**Comment by JanKneiphof** [comment]
> A little bit confusing: I can still start a "next page" even if there are no more corrections left. That allows me to cycle through empty pages.

Thank you for catching the bug with the next button!
This happened because the calculation for the `number-of-pages` was wrong in case there were no corrections available to display. This is fixed now.

**Review by hehin101 — APPROVED** [review]

File: src/clj/db.clj (hunk 2 above)
Comment (hehin101) [reply src/clj/db.clj]: I think the most import one is `[?user :user/id ?user-id] at the top. 👍

**Pull request merged**